ion-schema/src/test/kotlin/com/amazon/ionschema/reader/IonSchemaReaderV2_0Tests.kt

```diff
@@ -0,0 +1,324 @@
+                                    Constraint.CodepointLength(DiscreteIntRange(3))
+                                )
+                            ),
+                        )
+                    ),
+                    SchemaDocument.Item.Type(
+                        NamedTypeDefinition(
+                            typeName = "type2",
+                            typeDefinition = TypeDefinition(
+                                constraints = setOf(
+                                    Constraint.Type(TypeArgument.Reference("symbol")),
+                                    Constraint.Utf8ByteLength(DiscreteIntRange(5))
+                                )
+                            ),
+                        )
+                    )
+                )
+            ),
+            result.unwrap().first
+        )
+        assertFalse(result.unwrap().second.hasNext())
+    }
+
+    @Test
+    fun `readSchema can read a schema with a header`() {
+        val ionStream = ION.loader.load(
+            """
+            ${'$'}ion_schema_2_0
+            schema_header::{
+              imports: [
+                { id: "foo.isl" }
+              ],
+              user_reserved_fields: {
+                type: [foo],  
+              }
+            }
+        """
+        )
+
+        val result = reader.readSchema(ionStream)
+
+        assertTrue(result.isOk())
+        assertEquals(
+            SchemaDocument(
+                id = null,
+                ionSchemaVersion = IonSchemaVersion.v2_0,
+                items = listOf(
+                    SchemaDocument.Item.Header(
+                        imports = listOf(HeaderImport.Wildcard("foo.isl")),
+                        userReservedFields = UserReservedFields(type = setOf("foo"))
+                    ),
+                )
+            ),
+            result.unwrap().first
+        )
+        assertFalse(result.unwrap().second.hasNext())
+    }
+
+    @Test
+    fun `readSchema can read a schema with a footer and returns an iterator with all of the content after the footer`() {
+        val ionStream = ION.loader.load(
+            """
+            ${'$'}ion_schema_2_0
+            schema_footer::{}
+            abc
+            def
+            ${'$'}ion_schema_1_0
+        """
+        )
+
+        val result = reader.readSchema(ionStream)
+
+        assertTrue(result.isOk())
+        assertEquals(
+            SchemaDocument(null, IonSchemaVersion.v2_0, listOf(SchemaDocument.Item.Footer())),
+            result.unwrap().first
+        )
+        val iterator = result.unwrap().second
+        assertEquals(ION.singleValue("abc"), iterator.next())
+        assertEquals(ION.singleValue("def"), iterator.next())
+        assertEquals(ION.singleValue("\$ion_schema_1_0"), iterator.next())
+        assertFalse(iterator.hasNext())
+    }
+
+    @Test
+    fun `readSchema can read a schema with open content`() {
+        val ionStream = ION.loader.load(
+            """
```

Comment:
(Question) If we decide to return the contents that are not part of schema then should we return something that is before the version marker?
i.e. from the given test should we also return `foo` in the iterator?